### PR TITLE
MM-47102: Migrate "components/channel_notifications_modal/components/section_title.jsx" to Typescript

### PR DESCRIPTION
**Pull request by VishakhaPoonia**
#### Summary
Migrate "components/channel_notifications_modal/components/section_title.jsx" to Typescript

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/21067
  JIRA: https://mattermost.atlassian.net/browse/MM-47102

#### Related Pull Requests
NA

#### Screenshots
NA

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by AshishDhama — APPROVED** [review]

File: components/channel_notifications_modal/components/expand_view.jsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ import SettingItemMax from 'components/setting_item_max.jsx';
 
 import Describe from './describe.tsx';
 import ExtraInfo from './extra_info.jsx';
-import SectionTitle from './section_title.jsx';
+import SectionTitle from './section_title.tsx';
```
Comment (M-ZubairAhmed) [inline components/channel_notifications_modal/components/expand_view.jsx:17]: tsx ext is not needed

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: components/channel_notifications_modal/components/expand_view.jsx (hunk 1 above)
Comment (VishakhaPoonia) [reply components/channel_notifications_modal/components/expand_view.jsx:17]: @M-ZubairAhmed I think we cannot in JSX file
Screenshot 2022-10-05 at 9 48 59 PM

Comment (M-ZubairAhmed) [reply components/channel_notifications_modal/components/expand_view.jsx:17]: i am not able to get that issue thought locally i tested, the path resolves file. Our webpack config supports checking for tsx first and then later jsx. https://webpack.js.org/configuration/resolve/#resolveextensions
May be its the issue in your code editors eslint.

Comment (M-ZubairAhmed) [reply components/channel_notifications_modal/components/expand_view.jsx]: also apologies, i just pushed the commit to see if this error is caught in CI

**Review by M-ZubairAhmed — APPROVED** [review]

**Pull request merged**